### PR TITLE
ViewDocInfo: Better tab complete

### PR DESCRIPTION
**Pull request by critiqjo**
Test it out...

### CODE REVIEW

**Comment by powerman** [comment]
Thanks, this one works very nice. Now only two issues left:

- opening first node (in top dir, which is cpio for me and tar for you) instead of "not found" error
- possible improvement to make autocompletion case-insensitive

Second is optional, let me know if you're willing to do it. But first should be fixed before release. If you have no ideas why it happens and how to fix it - let me know, I'll try to fix it myself.

**Comment by powerman** [comment]
Also, it's probably good idea to change "Maintainer:" line in that plugin to your email, if you like.

**Comment by critiqjo** [comment]
The auto-completion is already case-insensitive.

Sorry, I don't have much idea about the cpio/tar problem.

My email is `john.ch.fr@gmail.com`

**Comment by powerman** [comment]
> The auto-completion is already case-insensitive.

Really? Then it doesn't work for me. How exactly you see this?

**Comment by critiqjo** [comment]
Try `:ViewDocInfo gawk<Tab>` ... I get `Gawkinet` and `Gawk` as candidates...

**Comment by powerman** [comment]
> Try :ViewDocInfo gawk<Tab> ... I get Gawkinet and Gawk as candidates...

Doesn't work for me. Maybe you forget to push some local modifications?

File: plugin/viewdoc_info.vim [hunk 1]
```diff
@@ -212,19 +212,25 @@ endfunction
...
+			let pipe = ' | sed -n ''s/\* [^:]*: \(([^.]*\)\..*/\1/p'' | sort | uniq'
+		endif
+	else
+		let pipe = ' | sed -e ''/^\* Menu:/,$ !d'' -n -e ''s/^\* \([^:]*\)::.*/\1/ p'''
+		let args = join(map(trail, 'shellescape(v:val)'), ' ')
+		let base_cmd = substitute(base_cmd, "'(dir)Top'", args, '')
 	endif
-	let part = part . (len(a:ArgLead) == 0 ? ' ' : '')
-	return escape(system(base_cmd . pipe . " | sed -n " . shellescape("/^" . part . "/p")), ' ')
+	return escape(system(base_cmd . pipe . " | sed -n " . shellescape("/^" . lead . "/Ip")), ' ')
```
Comment (critiqjo) [inline plugin/viewdoc_info.vim]: See `"/Ip"` part... `I` means ignore case... But someone mentions [here](http://stackoverflow.com/q/4412945/#comment31685516_4412964) that `I` is not a portable solution. May be something to do with your sed version?

Comment (powerman) [inline plugin/viewdoc_info.vim]: Actually `/I` works in my `sed`. I'll check this.
```sh
$ echo Test | sed -n '/^test/p'
$ echo Test | sed -n '/^test/Ip'
Test
```

Comment (powerman) [inline plugin/viewdoc_info.vim]: Got it. It works for you because you've `:set ignorecase`.

**Comment by powerman** [comment]
I've just fixed all known issues (af331d6b1632b9989259a52ff867466d810dc61b, 163e7f40b680607a4c83c08274e4a5fcbcfefb9f, 28a982d63683c2a631cd06f97dc88ec33cfc970f) - can you please take a look and test before I'll release it?

**Comment by critiqjo** [comment]
>It works for you because you've :set ignorecase.

Really?! So, after 163e7f4, does it work even without `ignorecase`?

`:ViewDocInfo Gzip` is not working for me... I've a fix, I'll issue a PR in an hour...

**Comment by powerman** [comment]
> So, after 163e7f4, does it work even without ignorecase?

Yep.

> :ViewDocInfo Gzip is not working for me... I've a fix, I'll issue a PR in an hour...

For me too. The info plugin is really overcomplicated, full of special cases… Is there really no better/simpler way to get info documentation without all these complications?

**Comment by critiqjo** [comment]
>Is there really no better/simpler way to get info documentation without all these complications?

It's because Gnu Info does not have predefined standards/layout. See 
```
$ info '(standards)'
$ info '(standards-2.13)'
$ info Standards
```
Only the second one correctly opens it for me. The original author made `s:FixNodeName` to address this issue. If Gnu Info handled such cases, we could have just used its output... Even Gnu Info doesn't know what to do?!!

**Pull request merged**